docs(world): introduction (in the new docs) (#1816)
latticexyz/mud: next-docs/pages/world/introduction.mdx

# Introduction

[`Store`](/store/introduction) provides low-level primitives for storage.
`World` provides the primitives for logic and access control on top of that storage.

Calls are sent to the `World` as a central entry point.
The `World` performs access control checks and if the request is authorized, sends it to the appropriate [`System`](/world/systems).

> Any access from the `System` to [the tables that store the data](/world/tables) also goes through the `World`, making it impossible to bypass access control.

Reviewer: There is one exception to this: root systems, which bypass access control by using the internal libraries. Do you think it's worth mentioning that here?

Those `System` contracts themselves are stateless, all the data in managed by the `World`.

A `World` is like the [diamond design pattern](https://eips.ethereum.org/EIPS/eip-2535) in many ways.
The big difference is permissionless extensions.
Anybody can claim an unused [namespace](/world/namespaces-access-control), and put into it tables and `System` contracts to [extend the world](/guides/extending-world).

![overall illustration](./world-intro.svg)

> 1. An entity (either an [externally owned account](https://ethereum.org/en/developers/docs/accounts/#types-of-account) or a contract) sends a request to the `World`.

Reviewer: ```suggestion
1. An account (either an [externally owned account](https://ethereum.org/en/developers/docs/accounts/#types-of-account) or a contract) sends a request to the `World`.
```

2. The `World` authorizes the request and sends it to the appropriate `System`.

> 3. The `System` can read information from all the `World`'s tables, and modify information in tables in its own namespace.

Reviewer: ```suggestion
3. The `System` can read information from all the `World`'s tables, and modify information in tables it has access to (by default those in its own namespace).
```

# Introduction

[`Store`](/store/introduction) provides low-level primitives for storage.
`World` provides the primitives for logic and access control on top of that storage.

Calls are sent to the `World` as a central entry point.

> The `World` performs access control checks and if the request is authorized, sends it to the appropriate [`System`](/world/systems).

Reviewer: ```suggestion
The `World` performs [access control checks](/world/namespaces-access-control) and if the request is authorized, sends it to the appropriate [`System`](/world/systems).
```

In most cases access from the `System` to [the tables that store the data](/world/tables) also goes through the `World`, making it impossible to bypass access control.
The `System` contracts themselves are stateless, all the data is managed by the `World`.

**Note:** A `System` in the root namespace _could_ bypass the access control.
However, only the owner of the root namespace (by default, the account that deployed the `World`) can deploy into the root namespace.

> If you want to prove that only an authorized `System` will ever be able to change table data, you can revoke all access to the root namespace and change the owner to `address(0)`.

Reviewer: ```suggestion
To prevent systems from being registered in the root namespace, the root namespace owner can [transfer ownership](/world/namespaces-access-control) to `address(0)`.
```

> A `World` is like the [diamond design pattern](https://eips.ethereum.org/EIPS/eip-2535) in many ways.

Reviewer: ```suggestion
`Systems` in a `World` are comparable to [`Facets` in the `Diamond` pattern](https://eips.ethereum.org/EIPS/eip-2535) in many ways.
```
